### PR TITLE
HB-06

### PR DESCRIPTION
**Pull request by Max-Pochepets**
One to One

- Models implemented: Shopping cart, Ticket
- DAO's created: Shopping cart, Ticket
- Service created: Shopping cart

### CODE REVIEW

File: src/main/java/cinema/shop/model/ShoppingCart.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shopping_cart")
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @OneToOne(cascade = CascadeType.MERGE)
```
Comment (vicire) [inline src/main/java/cinema/shop/model/ShoppingCart.java]: Optional - why do you use cascade here?

**Review by vicire — APPROVED** [review]

File: src/main/java/cinema/shop/model/ShoppingCart.java (hunk 1 above)
Comment (Max-Pochepets) [reply src/main/java/cinema/shop/model/ShoppingCart.java]: I didn't correct this from the time I was using `merge()` on shopping cart. I'll remove it, THX :)

File: src/main/java/cinema/shop/model/MovieSession.java [hunk 2]
```diff
@@ -17,12 +15,10 @@
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-    @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "movie_id")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/shop/model/MovieSession.java:21]: there is no need to remove `@JoinColumn` annotations)

File: src/main/java/cinema/shop/model/ShoppingCart.java (hunk 1 above)
Comment (KseniiaMakarova) [inline src/main/java/cinema/shop/model/ShoppingCart.java]: let's remove cascade here

File: src/main/java/cinema/shop/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    private TicketDao ticketDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticketDao.add(ticket);
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/shop/service/impl/ShoppingCartServiceImpl.java]: if there is no shopping cart for the user - we still save ticket to the db, is it ok?

File: src/main/java/cinema/shop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        ShoppingCart shoppingCartByUser = getByUser(shoppingCart.getUser());
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/shop/service/impl/ShoppingCartServiceImpl.java]: do you need this?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**